lib/active_merchant/billing/gateways/datatrans.rb

```diff
@@ -196,26 +231,36 @@ def headers
       def url(endpoint, options = {})
         case endpoint
         when 'settle', 'credit', 'cancel'
-          "#{test? ? test_url : live_url}#{options[:transaction_id]}/#{endpoint}"
+          "#{test? ? test_url : live_url}transactions/#{options[:transaction_id]}/#{endpoint}"
+        when 'tokenize'
+          "#{test? ? test_url : live_url}aliases/#{endpoint}"
+        when 'delete_alias'
+          "#{test? ? test_url : live_url}aliases/#{options[:alias_id]}"
         else
-          "#{test? ? test_url : live_url}#{endpoint}"
+          "#{test? ? test_url : live_url}transactions/#{endpoint}"
         end
       end
 
       def success_from(action, response)
         case action
         when 'authorize', 'credit'
-          true if response.include?('transactionId') && response.include?('acquirerAuthorizationCode')
+          response.include?('transactionId') && response.include?('acquirerAuthorizationCode')
         when 'settle', 'cancel'
-          true if response.dig('response_code') == 204
+          response.dig('response_code') == 204
+        when 'tokenize'
+          response['responses'][0].include?('alias') && response['overview']['failed'] == 0
```

Comment:
💬 Comment/Readability:

just matter of taste but `has_key?` is more explicit for hashes than `include?` which is normally asociated with checking arrays elements.

Reply:
maybe more robust conditional:

```ruby
response.dig('responses', 0, 'alias') && response.dig('overview', 'failed') == 0
```